validator/validator_test.go

```diff
@@ -0,0 +1,71 @@
+package validator
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestValidateJSONFormatFailsForInvalidJSON(t *testing.T) {
+	err := ValidateJSONFormat(`["connection lost", "invalid request"]`)
+	assert.Error(t, err)
+}
+
+func TestValidateJSONFormatFailsForMissingRequiredFields(t *testing.T) {
```

Comment:
Add log to explain why? e.g.

`t.Log("expect error if `source` field is missing")`